### PR TITLE
add images to product collections

### PR DESCRIPTION
**Pull request by edihasaj**
add thumbnail and image properties to product-collection.ts.
add api endpoint changes for managing thumbnail and image.
add migration changes for product collection images
add new endpoint for querying collection with products and images included (joined) /store/collections/:id/products
add integration test for new created endpoint.
add docs changes for product-collection entity changes
add docs for the new added endpoint.

### CODE REVIEW

**Comment by edihasaj** [comment]
@srindom Is this okay?

File: packages/medusa/src/services/product-collection.js [hunk 1]
```diff
@@ -107,15 +107,47 @@ class ProductCollectionService extends BaseService {
...
+      const { images, ...rest } = collection
+      if (!rest.thumbnail && images && images.length) {
+        rest.thumbnail = images[0]
+      }
+
       try {
         const productCollection = await collectionRepo.create(collection)
+
+        if (images) {
+          productCollection.images = await this.upsertImages_(images)
```
Comment (kasperkristensen) [inline packages/medusa/src/services/product-collection.js:119]: You are re-implementing a method that already exists in the ImageRepository. 

Instead, you could just do like this:

```ts
const imageRepo = manager.getCustomRepository(this.imageRepository_)

// ...

if (images?.length) {
  productCollection.images = await imageRepo.upsertImages_(images)
}
```

File: packages/medusa/src/services/product-collection.js [hunk 2]
```diff
@@ -107,15 +107,47 @@ class ProductCollectionService extends BaseService {
...
+        }
+
         return await collectionRepo.save(productCollection)
       } catch (error) {
         throw formatException(error)
       }
     })
   }
 
+  async upsertImages_(images) {
```
Comment (kasperkristensen) [inline packages/medusa/src/services/product-collection.js:129]: No need for this method, as explained above.

File: packages/medusa/src/services/product-collection.js [hunk 3]
```diff
@@ -128,9 +160,19 @@ class ProductCollectionService extends BaseService {
...
-      const { metadata, ...rest } = update
+      const { metadata, images, ...rest } = update
+
+      if (!collection.thumbnail && !update.thumbnail && images?.length) {
+        collection.thumbnail = images[0]
+      }
+
+      if (images) {
+        collection.images = await this.upsertImages_(images)
+      }
```
Comment (kasperkristensen) [inline packages/medusa/src/services/product-collection.js:175]:
```suggestion
     // at top of method
     const imageRepo = this.manager_.getCustomRepository(
          this.imageRepository_
      )
    
      if (images) {
          collection.images = await imageRepo.upsertImages_(images)
      }
```

File: packages/medusa/src/services/product-collection.js [hunk 4]
```diff
@@ -107,15 +107,47 @@ class ProductCollectionService extends BaseService {
         this.productCollectionRepository_
       )
 
+      const { images, ...rest } = collection
+      if (!rest.thumbnail && images && images.length) {
```
Comment (kasperkristensen) [inline packages/medusa/src/services/product-collection.js:111]: nit

```suggestion
      if (!rest.thumbnail && images?.length) {
```

**Review by kasperkristensen — CHANGES_REQUESTED** [review]
I very much like the idea of adding images to collections! We could take it further by adding an optional `description`.  

I've added a couple of comments :)

**Comment by edihasaj** [comment]
I am also developing the remaining part in integrating medusa admin with the added functionality above.

**Comment by edihasaj** [comment]
Requested functionality for querying imageRepository from DI, doesn't recognize the upsertImages() method, so I'm converting the services/product-collection.js to typescript. I'll test it and I'll let  you know for review continutation

**Comment by edihasaj** [comment]
@kasperkristensen you can now review the changes.

**Pull request closed without merging**